PreFreeSurfer/PreFreeSurferPipeline.sh

```diff
@@ -452,253 +517,312 @@ log_Msg "POSIXLY_CORRECT="${POSIXLY_CORRECT}
+
+    # Remove the file (warpfield) that serves as a proxy in FreeSurferPipeline for whether PostFreeSurfer has been run
+    # i.e., whether the T1w/T2w_acpc_dc* volumes reflect the PreFreeSurferPipeline versions (above)
+    # or the PostFreeSurferPipeline versions
+    OutputOrigT2wToT1wPostFS=OrigT2w2T1w  #Needs to match name used in both FreeSurferPipeline and PostFreeSurferPipeline
+    imrm ${OutputOrigT2wToT1wPostFS}
+  fi
 
-# ------------------------------------------------------------------------------
-#  Bias Field Correction: Calculate bias field using square root of the product
-#  of T1w and T2w iamges.
-# ------------------------------------------------------------------------------
 
-log_Msg "Performing Bias Field Correction"
-if [ ! -z ${BiasFieldSmoothingSigma} ] ; then
-	BiasFieldSmoothingSigma="--bfsigma=${BiasFieldSmoothingSigma}"
-fi
+# -- Are we using a custom mask?
 
-log_Msg "mkdir -p ${T1wFolder}/BiasFieldCorrection_sqrtT1wXT1w"
-mkdir -p ${T1wFolder}/BiasFieldCorrection_sqrtT1wXT1w
-
-${RUN} ${HCPPIPEDIR_PreFS}/BiasFieldCorrection_sqrtT1wXT1w.sh \
-	--workingdir=${T1wFolder}/BiasFieldCorrection_sqrtT1wXT1w \
-	--T1im=${T1wFolder}/${T1wImage}_acpc_dc \
-	--T1brain=${T1wFolder}/${T1wImage}_acpc_dc_brain \
-	--T2im=${T1wFolder}/${T2wImage}_acpc_dc \
-	--obias=${T1wFolder}/BiasField_acpc_dc \
-	--oT1im=${T1wFolder}/${T1wImage}_acpc_dc_restore \
-	--oT1brain=${T1wFolder}/${T1wImage}_acpc_dc_restore_brain \
-	--oT2im=${T1wFolder}/${T2wImage}_acpc_dc_restore \
-	--oT2brain=${T1wFolder}/${T2wImage}_acpc_dc_restore_brain \
-	${BiasFieldSmoothingSigma}
+elif [ "$CustomBrain" = "MASK" ] ; then
 
-# ------------------------------------------------------------------------------
-# Create a one-step resampled version of the {T1w,T2w}_acpc_dc outputs
-# (applied after GDC, which we don't bundle in, because of the possible need
-# to average multiple T1/T2 inputs).
-
-# This overwrites the {T1w,T2w}_acpc_dc outputs created above, and mimics what
-# occurs at the beginning of PostFreeSurfer/CreateMyelinMaps.sh.
-# Note that the CreateMyelinMaps equivalent is still needed though because
-# (1) T1w_acpc_dc_restore_brain is (re)generated with a better estimate of
-#     the brain mask, provided by FreeSurfer
-# (2) the entire set of T2w_acpc_dc outputs needs to be regenerated, using the
-#     refinement to the "T2wtoT1w" registration that FreeSurfer provides.
-
-# Just implement inline, rather than writing a separate script
-# Added 2/19/2019
-# ------------------------------------------------------------------------------
+  log_Msg "Skipping all the steps to Atlas registration, applying custom mask."
+  verbose_red_echo "---> Applying custom mask"
 
-log_Msg "Creating one-step resampled version of {T1w,T2w}_acpc_dc outputs"
+  ${FSLDIR}/bin/fslmaths ${T1wFolder}/${T1wImage}_acpc_dc_restore -mas ${T1wFolder}/custom_acpc_dc_restore_mask ${T1wFolder}/${T1wImage}_acpc_dc_restore_brain
```

Comment:
In this particular script, we simply call using `fslmaths` rather than `${FSLDIR}/bin/fslmaths`.  (Yes, we know we are are terribly inconsistent on that front, but I believe that the longer term goal is to eliminate the inclusion of environment variables within the script for setting tool paths).

Reply:
Ok